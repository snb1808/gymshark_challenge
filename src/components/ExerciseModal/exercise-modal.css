.modal-button {
    background-color: darkgrey;
    border-color: #888888;
}

.modal-button:hover {
    border-color: #888888;
    background-color: dimgray;
}

.modal-button:focus {
    border-color: #888888;
    background-color: dimgray;
}