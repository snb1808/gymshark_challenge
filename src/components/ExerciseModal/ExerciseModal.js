import { Modal, Button } from 'react-bootstrap';
import './exercise-modal.css';

const ExerciseModal = (props) => {
    return (
        <Modal
            {...props}
            size="lg"
            aria-labelledby='modal-title'
            centered
            backdrop='static'
            keyboard={false}
            aria-describedby='description'
        >
            <Modal.Header closeButton>
                <Modal.Title id='modal-title' tabindex='0'>
                    {props.content.name}
                </Modal.Title>
            </Modal.Header>
            <Modal.Body>
                <h4>Directions:</h4>
                <p id='description' dangerouslySetInnerHTML={{__html: props.content.transcript}}/>
            </Modal.Body>
            <Modal.Footer>
                <Button className='modal-button'>OK</Button>
            </Modal.Footer>
        </Modal>
    );
};

export default ExerciseModal;