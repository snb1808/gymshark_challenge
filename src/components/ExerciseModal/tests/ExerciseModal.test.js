import { shallow } from 'enzyme';
import ExerciseModal from '../ExerciseModal';

describe('ExerciseModal', () => {
    it('should render correctly', () => {
        const props = {
            content: {
                name: 'Exercise name',
                transcript: 'Exercise transcript',
            },
        };

        const component = shallow(<ExerciseModal {...props} />);

        expect(component.debug()).toMatchSnapshot();
    });
});