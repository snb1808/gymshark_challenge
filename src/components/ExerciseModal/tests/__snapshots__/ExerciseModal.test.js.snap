// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExerciseModal should render correctly 1`] = `
"<Modal content={{...}} size=\\"lg\\" aria-labelledby=\\"modal-title\\" centered={true} backdrop=\\"static\\" keyboard={false} aria-describedby=\\"description\\" show={false} autoFocus={true} enforceFocus={true} restoreFocus={true} animation={true} dialogAs={{...}}>
  <ModalHeader closeButton={true} closeLabel=\\"Close\\">
    <ModalTitle id=\\"modal-title\\" tabindex=\\"0\\">
      Exercise name
    </ModalTitle>
  </ModalHeader>
  <ModalBody>
    <h4>
      Directions:
    </h4>
    <p id=\\"description\\" dangerouslySetInnerHTML={{...}} />
  </ModalBody>
  <ModalFooter>
    <Button className=\\"modal-button\\" variant=\\"primary\\" active={false} disabled={false}>
      OK
    </Button>
  </ModalFooter>
</Modal>"
`;
